components/engine/src/rocks/snapshot.rs

```diff
@@ -60,6 +60,14 @@ impl Snapshot {
         }
         Ok(DBIterator::new_cf(Arc::clone(&self.db), handle, opt))
     }
+
+    pub fn get_sequence_number(&self) -> u64 {
```

Comment:
em, I remember Snapshot now has a get sequence number API, you don't need to call from FFI

Reply:
The Snapshot struct there doesn't use the Snapshot struct of our rocksdb-rust crate, so I have to add a method there.
I think it needs to be refactored by using the Snapshot struct of rocksdb-rust directly.

Reply:
Or we can add the `get_sequence_number` method to the `UnsafeSnap` struct.